app/controllers/concerns/hal_actions/errors.rb

```diff
@@ -13,6 +13,16 @@ def status
     end
   end
 
+  class NotFound < StandardError
+    def message
+      'Resource not found'
+    end
+
+    def status
+      404
+    end
+  end
+
```

Comment:
Sounds good.  I added this to catch some activerecord-not-found errors in very specific places, for the show/update/destroy actions.  Hopefully specific enough that other unrelated `model.find()` errors won't trigger an incorrect 404.